### PR TITLE
Addpost

### PR DESCRIPTION
**Pull request by praveensree**

### CODE REVIEW

File: Server/Controllers/PostsController.cs [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Server.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class PostsController : ControllerBase
```
Comment (akshayverma91) [inline Server/Controllers/PostsController.cs:15]: Should be singular not plural like Post not Posts
Also be specific, it seems its generic Post should be like NewsPost or SocialPost

File: Server/Controllers/PostsController.cs [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    {
+        // GET: api/<PostsController>
+        [HttpGet]
+        public IActionResult GetAllPosts()
```
Comment (akshayverma91) [inline Server/Controllers/PostsController.cs:19]: Change here GetSocialPost

File: Server/Controllers/PostsController.cs [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+            }
+            catch (Exception)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+        }
+
+        // GET api/<PostsController>/5
+        [HttpGet("{id}")]
+        public IActionResult GetPostsById(int id)
```
Comment (akshayverma91) [inline Server/Controllers/PostsController.cs:37]: Change here GetSocialPostById

File: Server/Controllers/PostsController.cs [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+            }
+            catch (Exception)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+        }
+
+        // POST api/<PostsController>
+        [HttpPost]
+        public IActionResult AddNewPost([FromBody] Post post)
```
Comment (akshayverma91) [inline Server/Controllers/PostsController.cs:55]: Change here CreateSocialPost

File: Server/Controllers/PostsController.cs [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+            }
+            catch (Exception)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+        }
+
+        // PUT api/<PostsController>/5
+        [HttpPut("{id}")]
+        public IActionResult ModifyExistingPost(int id, [FromBody] Post post)
```
Comment (akshayverma91) [inline Server/Controllers/PostsController.cs:78]: Change here UpdateSocialPost

File: Server/Controllers/PostsController.cs [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+                    return Ok(fb.Posts.First(x => x.PostId == id));
+                }
+            }
+            catch (Exception)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+        }
+        [HttpPut("likes/{id}")]
+        public IActionResult AddLike(int id, [FromBody] Post post)
```
Comment (akshayverma91) [inline Server/Controllers/PostsController.cs:98]: Change here UpdateSocialPostLike

File: Server/Controllers/PostsController.cs [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+                    return Ok(update.Likes);
+                }
+            }
+            catch (Exception)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+        }
+        [HttpPut("dislikes/{id}")]
+        public IActionResult UnLike(int id, [FromBody] Post post)
```
Comment (akshayverma91) [inline Server/Controllers/PostsController.cs:116]: this should be present and one api endpoint will serve it purpose to update like or heart counter, so remove this

File: Server/Controllers/PostsController.cs [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+                }
+            }
+            catch (Exception)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+        }
+
+        [HttpPut("hearts/{id}")]
+        public IActionResult AddHeart(int id, [FromBody] Post post)
```
Comment (akshayverma91) [inline Server/Controllers/PostsController.cs:140]: Same as like comment above

File: Server/Controllers/PostsController.cs [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+                }
+            }
+            catch (Exception)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+        }
+
+        [HttpPut("dishearts/{id}")]
+        public IActionResult RemoveHeart(int id, [FromBody] Post post)
```
Comment (akshayverma91) [inline Server/Controllers/PostsController.cs:159]: same as like api endpoint

File: Server/WeatherForecast.cs [hunk 10]
```diff
@@ -0,0 +1,15 @@
+using System;
+
+namespace Server
+{
+    public class WeatherForecast
```
Comment (akshayverma91) [inline Server/WeatherForecast.cs:5]: delete this file not needed in as per out task

**Review by akshayverma91 — CHANGES_REQUESTED** [review]

File: Server/Controllers/PostsController.cs [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+            }
+            catch (Exception)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+        }
+
+        // DELETE api/<PostsController>/5
+        [HttpDelete("{id}")]
+        public IActionResult RemovePost(int id)
```
Comment (akshayverma91) [inline Server/Controllers/PostsController.cs:186]: delete this endpoint as per story we don't have delete post

**Comment by akshayverma91** [comment]
@praveensree also add api endpoint for total like and total heart count

File: Server/Controllers/PostsController.cs (hunk 1 above)
Comment (praveensree) [reply Server/Controllers/PostsController.cs]: Changed the controller name to SocialPost Controller

File: Server/Controllers/PostsController.cs (hunk 2 above)
Comment (praveensree) [reply Server/Controllers/PostsController.cs]: changed to GetSocialPost

File: Server/Controllers/PostsController.cs (hunk 3 above)
Comment (praveensree) [reply Server/Controllers/PostsController.cs]: changed to GetSocialPostById

File: Server/Controllers/PostsController.cs (hunk 4 above)
Comment (praveensree) [reply Server/Controllers/PostsController.cs]: changed to CreateSocialPost

File: Server/Controllers/PostsController.cs (hunk 5 above)
Comment (praveensree) [reply Server/Controllers/PostsController.cs]: changed to UpdateSocialPost

File: Server/Controllers/PostsController.cs (hunk 6 above)
Comment (praveensree) [reply Server/Controllers/PostsController.cs]: changed to UpdateSocialPostLike

File: Server/Controllers/PostsController.cs (hunk 7 above)
Comment (praveensree) [reply Server/Controllers/PostsController.cs]: Removed the method

File: Server/Controllers/PostsController.cs (hunk 8 above)
Comment (praveensree) [reply Server/Controllers/PostsController.cs]: changed to UpdateSocialPostHeart

File: Server/Controllers/PostsController.cs (hunk 9 above)
Comment (praveensree) [reply Server/Controllers/PostsController.cs]: Removed

File: Server/Controllers/PostsController.cs (hunk 11 above)
Comment (praveensree) [reply Server/Controllers/PostsController.cs]: deleted the endpoint

File: Server/WeatherForecast.cs (hunk 10 above)
Comment (praveensree) [reply Server/WeatherForecast.cs]: deleted the weatherforecast.cs file

**Pull request merged**